ZohoCRMCharts/Classes/ZCRMComparator.swift

```diff
@@ -0,0 +1,637 @@
+		cell.layer.borderWidth = 1
+		cell.layer.cornerRadius = 5
+		cell.layer.borderColor = chunk.color.cgColor
+		if !self.chunkDatas.isEmpty {
+			cell.chunkData = self.chunkDatas[index]
+			cell.highValue = self.getMaxOfChunk(index)
+		}
+	}
+	
+	private func renderElegantComparatorCell(_ cell: ZCRMComparatorCell, index: Int) {
+		
+		let group = self.groupings.groups[self.getGroupIndex(chunkDataIndex: index)]
+		cell.backgroundColor = group.bgColor
+		if !self.chunkDatas.isEmpty {
+			cell.chunkData = self.chunkDatas[index]
+		}
+	}
+	
+	private func renderClassicComparatorCell(_ cell: ZCRMComparatorCell, index: Int) {
+		
+		let cellIndex = index - self.chunks.count
+		if index < self.chunks.count {
+			cell.isHeader = true
+			cell.chunkData = ZCRMChunkData(label: self.chunks[index].label, value: 0)
+			cell.backgroundColor = self.renderOptions.classicHeaderRowColor
+		} else if !self.chunkDatas.isEmpty {
+			cell.chunkData = self.chunkDatas[cellIndex]
+		}
+		let totalCells = (self.groupings.groups.count + 1) * self.chunks.count
+		if index < totalCells - self.chunks.count {
+			cell.addBottomBorder(color: .black, width: 1)
+		}
+	}
+}
+
+// methods related to data source
+fileprivate extension ZCRMComparator {
+	
+	fileprivate func setChunkDatas() {
+		
+		if self.type == .classic {
+			for (groupIndex, group) in self.groupings.groups.enumerated() {
+				for (chunkIndex, chunk) in self.chunks.enumerated() {
+					self.chunkDatas.append(self.dataSource.comparator(chunk, group, chunkIndex: chunkIndex, groupIndex: groupIndex))
+				}
+			}
+		} else {
+			for (chunkIndex, chunk) in self.chunks.enumerated() {
+				for (groupIndex, group) in self.groupings.groups.enumerated() {
+					self.chunkDatas.append(self.dataSource.comparator(chunk, group, chunkIndex: chunkIndex, groupIndex: groupIndex))
+				}
+			}
+			self.reloadGroupImages()
+		}
+	}
+	
+	fileprivate func setGroupImages() {
+		
+		for (index, group) in self.groupings.groups.enumerated() {
+			
+			self.dataSource.groupImage(of: group) { (image) in
+				if let image = image {
+					self.setImageFor(index, image: image)
+				}
+			}
+		}
+	}
+	
+	private func setImageFor(_ groupAtIndex: Int, image: UIImage) {
+		
+		if self.headersView.arrangedSubviews.count > groupAtIndex {
+			(self.headersView.arrangedSubviews[groupAtIndex] as! ZCRMComparatorHeader).group.image = image
+		}
+	}
+	
+	fileprivate func getMaxOfChunk(_ indexInArray: Int) -> Int {
+		
+		let chunksLength = self.groupings.groups.count
+		let chunksRange = (indexInArray + 1).toCGFloat() / chunksLength.toCGFloat()
+		let chunkIndex = chunksRange.rounded(.up)
+		let chunkEndIndex = (chunkIndex * chunksLength.toCGFloat()) - 1
+		let chunkStartIndex = (chunkEndIndex - (chunksLength - 1).toCGFloat())
+		var chunkValues: [Int] = []
+		for index in chunkStartIndex.toInt()...chunkEndIndex.toInt() {
+			chunkValues.append(self.chunkDatas[index].value)
+		}
+		if chunkValues.isEmpty {
+			return 0
+		}
+		return chunkValues.max()!
```

Comment:
Avoid force un wrapping.

Reply:
Here It needed because the method providing optional value. And if there is no value we are returning zero in the previous. So it won't crash.